OpenCV searches for Python 2 first and only uses Python 3 during the build as a fallback.
Just remove the code searching for Python 2 to avoid it picking up a system Python.

Author: Alexander Grund (TU Dresden)

diff -ur a/cmake/OpenCVDetectPython.cmake b/cmake/OpenCVDetectPython.cmake
--- a/cmake/OpenCVDetectPython.cmake	2025-04-10 11:26:30.171363156 +0200
+++ b/cmake/OpenCVDetectPython.cmake	2025-04-10 11:27:13.896674205 +0200
@@ -268,14 +268,6 @@
   return()
 endif()

-find_python("" "${MIN_VER_PYTHON2}" PYTHON2_LIBRARY PYTHON2_INCLUDE_DIR
-    PYTHON2INTERP_FOUND PYTHON2_EXECUTABLE PYTHON2_VERSION_STRING
-    PYTHON2_VERSION_MAJOR PYTHON2_VERSION_MINOR PYTHON2LIBS_FOUND
-    PYTHON2LIBS_VERSION_STRING PYTHON2_LIBRARIES PYTHON2_LIBRARY
-    PYTHON2_DEBUG_LIBRARIES PYTHON2_LIBRARY_DEBUG PYTHON2_INCLUDE_PATH
-    PYTHON2_INCLUDE_DIR PYTHON2_INCLUDE_DIR2 PYTHON2_PACKAGES_PATH
-    PYTHON2_NUMPY_INCLUDE_DIRS PYTHON2_NUMPY_VERSION)
-
 option(OPENCV_PYTHON3_VERSION "Python3 version" "")
 find_python("${OPENCV_PYTHON3_VERSION}" "${MIN_VER_PYTHON3}" PYTHON3_LIBRARY PYTHON3_INCLUDE_DIR
     PYTHON3INTERP_FOUND PYTHON3_EXECUTABLE PYTHON3_VERSION_STRING
